src/drivers/imu/mpu9250/mpu9250.cpp

```diff
@@ -1456,8 +1481,15 @@ MPU9250::check_registers(void)
 		if (_register_wait == 0 || _checked_next == 0) {
 			// if the product_id is wrong then reset the
 			// sensor completely
-			write_reg(MPUREG_PWR_MGMT_1, BIT_H_RESET);
-			write_reg(MPUREG_PWR_MGMT_2, MPU_CLK_SEL_AUTO);
+
+		    if(_device_type == MPU_DEVICE_TYPE_ICM20948) {
+		        // reset_mpu();
+		    }
+		    else {
+			    write_reg(MPUREG_PWR_MGMT_1, BIT_H_RESET);
```

Comment:
I doubt if this is working for the MPU9250. It just resets the chip, but doesn't set it up again?